### PR TITLE
Instructor help page: add side navigation #5616

### LINKED ISSUE #5616: Instructor help page: add side navigation
Currently, we are using the "back to top" method for users to navigate back to the contents listing, this can get quite tedious after a while.

Having a side navigation (something like the [bootstrap components page](http://getbootstrap.com/components/)) would make the help page more helpful.

### PR DESCRIPTION
**Pull request by amarlearning**
Fixes #5616

**Outline of Solution**

As there are a couple of ways to add a side navigation bar for help, this is one of the simplest way to do that. Let me know how complex can I go or if this needs to modified or any changes needed in this.
_[Screenshot]_
screenshot from 2017-06-08 05-25-53

### CODE REVIEW

**Comment by amarlearning** [comment]
@li-kai I was hoping for a completely different reply.
Though there are a couple of ways to make this better, I would like to bring them in front of you.

1. We can use the default bootstrap `Scrollspy` component. The example of how Scrollspy works. In [this](http://getbootstrap.com/getting-started/) page, see how the side navigation updates itself with the current heading.

2. Another thing that we can do is to make a dropdown menu in the side section with sub-heading as well. As of now, i have only added headings in the side section.

Let me know your thoughts on this.

**Comment by whipermr5** [comment]
@li-kai Reminder :)

**Comment by VamsiSangam** [comment]
Working on `instructorHelp` myself, makes me eager for this feature 😝 , scrolling up and down is pretty tedious.

One advice, can't we remove the old top navigation, because we getting the side navigation method? And regarding mobile view, maybe we should keep the side navigation hidden just as [Bootstrap](http://getbootstrap.com/getting-started/) does.

**Comment by whipermr5** [comment]
@amarlearning You can go as complex as you want to and we'll take a look at the final results :) I'm  not much of a UI expert but will try my best to review. @VamsiSangam feel free to offer suggestions too :)

**Comment by amarlearning** [comment]
@whipermr5 Thanks. I'll update my PR asap.

**Comment by HirdayGupta** [comment]
@amarlearning are you still working on this?

**Comment by amarlearning** [comment]
@hirdaygupta yes, I am working on this. I kept it aside for a while because I was waiting for Vamsi PR to get merged.

**Comment by wkurniawan07** [comment]
@amarlearning leaving a note here that the linked PR has been merged.

**Comment by whipermr5** [comment]
Hey @amarlearning, you still working on this? :)

**Comment by amarlearning** [comment]
Yes, I will update my PR ASAP. 
Sorry for the delay 😅

**Comment by amarlearning** [comment]
Yes, I will update my PR ASAP. 
Sorry for the delay 😅

**Comment by amarlearning** [comment]
Appologies for the delay, ready for review. [Screenshot added]

File: src/main/webapp/stylesheets/teammatesCommon.css [hunk 1]
```diff
@@ -617,12 +617,64 @@ input.inline {
...
+    }
+
+    .helpGuideData {
+        width: 100%;
+    }
+
+    .helpGuideSideBar {
+        display: none;
+    }
+
```
Comment (whipermr5) [inline src/main/webapp/stylesheets/teammatesCommon.css]: Remove this blank line to fix the build error.

**Comment by amarlearning** [comment]
@whipermr5 I don't have an international debit card, so can't host my application online. _[GIF added]_

out-1 1

**Comment by amarlearning** [comment]
Anyone please help me with the build issue

**Comment by whipermr5** [comment]
@amarlearning You need to update [this](https://github.com/TEAMMATES/teammates/blob/ecb3ee54172f1c832069cbbba07c412f2f157d4e/src/test/java/teammates/test/pageobjects/InstructorHelpPage.java#L11) line.

**Comment by amarlearning** [comment]
Ready for review

File: src/main/webapp/instructorHelp.jsp [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <%@ taglib tagdir="/WEB-INF/tags" prefix="t" %>
 <t:helpPage>
-    <h1>Help for Instructors</h1>
+    <h1 class="HelpHeading">Help for Instructors</h1>
```
Comment (li-kai) [inline src/main/webapp/instructorHelp.jsp]: Keep the class name to camelCase. And all's good!

**Comment by amarlearning** [comment]
ready for review

**Comment by whipermr5** [comment]
Many of the UI elements aren't very responsive, and the reduction in width has visibly negative consequences:
screen shot 2017-06-15 at 7 03 32 pm
screen shot 2017-06-15 at 6 59 44 pm

Originals:
screen shot 2017-06-15 at 7 03 34 pm
screen shot 2017-06-15 at 6 59 47 pm

However they aren't really your fault. Perhaps this should be put on hold until those issues are fixed?

**Comment by amarlearning** [comment]
Looks like the problem that we are facing over here is same as #7467, If one can resolve issue #7467 problem that we are facing over here will be solved. Taking issue #7467

**Comment by wkurniawan07** [comment]
Closing due to an inactivity for a very long period. If anyone is interested to continue the work done here, feel free to reopen/request for reopen.

**Pull request closed without merging**